### PR TITLE
logrotate.conf: fixes for systemd

### PR DESCRIPTION
**Pull request by smithfarm**
Before this patch, the command 'logrotate -f /etc/logrotate.d/ceph'
was generating an error "Failed to reload ceph.target: Job type reload is not
applicable for unit ceph.target".

Before we issue systemctl reload, check that there is at least
one active ceph-* service.

Since we use grep, make the grep package a dependency.

http://tracker.ceph.com/issues/12173 Fixes: #12173

Signed-off-by: Tim Serong <tserong@suse.com>
Signed-off-by: Lars Marowsky-Bree <lmb@suse.com>
Signed-off-by: Nathan Cutler <ncutler@suse.com>

### CODE REVIEW

**Comment by dachary** [comment]
@smithfarm I know nothing of systemd and probably not the best person to review what looks like a very fine commit.

**Comment by smithfarm** [comment]
I can't fathom why the test is failing in this way. I can see that the test is calling "ceph-disk activate" which then (apparently) calls ceph-osd, which returns -15:

    subprocess.CalledProcessError: Command '['ceph-osd', '--cluster', 'ceph', 
    '--mkfs', '--mkkey', '-i', '0', '--monmap', 'testdir/osd-config/0/activate.monmap', 
    '--osd-data', 'testdir/osd-config/0', '--osd-journal', 'testdir/osd-config/0/journal', 
    '--osd-uuid', '3e7bad3f-8c36-4fdf-a2d3-33c77798598c', '--keyring', 
    'testdir/osd-config/0/keyring']' returned non-zero exit status -15

At that point, the trail goes cold for me.

**Comment by smithfarm** [comment]
running it through the make check bot one more time

**Comment by smithfarm** [comment]
also running make check locally in EL7 docker container

File: src/logrotate.conf [hunk 1]
```diff
@@ -8,6 +8,10 @@
             service ceph reload >/dev/null
         elif which invoke-rc.d > /dev/null 2>&1 && [ -x `which invoke-rc.d` ]; then
             invoke-rc.d ceph reload >/dev/null
+        elif which systemctl > /dev/null 2>&1 && [ -x `which systemctl` ]; then
+            if systemctl is-active 'ceph-*' | grep -i active > /dev/null 2>&1 ; then 
+                systemctl reload 'ceph-*' >/dev/null
```
Comment (ktdreyer) [inline src/logrotate.conf]: Do we care about the exit code of `systemctl reload` here? If not we could tack on "`|| :` . (just looking at what other packages do)

**Comment by ktdreyer** [comment]
Looks good to me! @dachary mind confirming that the make check bot failure isn't related to this change?

File: src/logrotate.conf (hunk 1 above)
Comment (smithfarm) [inline src/logrotate.conf]: Do you mean `systemctl reload 'ceph-*' || :` instead of `systemctl reload 'ceph-*' >/dev/null` ?

Comment (smithfarm) [inline src/logrotate.conf]: or `systemctl reload 'ceph-*' >/dev/null || :` ?

Comment (smithfarm) [inline src/logrotate.conf]: OK, now I see that the point is to ensure that we don't leave `$?` set to false for no good reason. So the latter. Will do.

**Comment by smithfarm** [comment]
I ran make check in a CentOS 7 docker container and `PASS: test/osd/osd-config.sh` so it would seem the bot failure is "environmental noise"?

**Comment by smithfarm** [comment]
Local `make check` completed successfully. @ktdreyer `|| :` tacked on.

**Comment by ktdreyer** [comment]
Ok, let's merge this and assume the make check bot failure is spurious :)

**Pull request merged**